src/Microsoft.ML.Data/Evaluators/RegressionEvaluator.cs

```diff
@@ -102,6 +102,10 @@ public override double RSquared
                 {
                     get
                     {
+                        // RSquared value cannot be well-defined with less than two samples.
+                        // Return NaN instead of -Infinity.
+                        if ((TotalLabelSquaredW - TotalLabelW * TotalLabelW / SumWeights) == 0)
+                            return double.NaN;
```

Comment:
Could you add a test for this?

Reply:
There isn't really a direct way of testing this apart from forcing a cross validation scenario in AutoML where the size of the folds in cross validation is less than 2. If a test for this is necessary then I will go this route.

Reply:
Why is AutoML required for this? 

---
In reply to: [459046236](https://github.com/dotnet/machinelearning/pull/5319#discussion_r459046236) [](ancestors = 459046236)

Reply:
My bad on wording, AutoML isn't required to test this, the original issue exposed this bug through an AutoML experiment. A call to `mlContext.Regression.Evaluate(prediction)` where `prediction` is a one-row IDataView would be enough to test this.

Reply:
Could make a test which trains a regression model then tests on one row of data, checking that the output is `NaN` for R2.

Reply:
Added test and completed the request change to merge PR.